x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/bulk_actions/use_bulk_actions.tsx

```diff
@@ -116,28 +120,31 @@ export const useBulkActions = ({
           ? disabledRules.map(({ id }) => id)
           : disabledRulesNoML.map(({ id }) => id);
 
-        await executeRulesBulkAction({
+        const res = await executeRulesBulkAction({
           visibleRuleIds: ruleIds,
           action: BulkAction.enable,
           setLoadingRules,
           toasts,
           search: isAllSelected ? { query: filterQuery } : { ids: ruleIds },
         });
-        invalidateRules();
+        updateRulesCache(res?.attributes?.results?.updated ?? []);
+        setIsRefreshOn(true);
       };
 
       const handleDisableActions = async () => {
         startTransaction({ name: BULK_RULE_ACTIONS.DISABLE });
         closePopover();
         const enabledIds = selectedRules.filter(({ enabled }) => enabled).map(({ id }) => id);
-        await executeRulesBulkAction({
+
+        const res = await executeRulesBulkAction({
           visibleRuleIds: enabledIds,
           action: BulkAction.disable,
           setLoadingRules,
           toasts,
           search: isAllSelected ? { query: filterQuery } : { ids: enabledIds },
         });
-        invalidateRules();
+        updateRulesCache(res?.attributes?.results?.updated ?? []);
+        setIsRefreshOn(true);
```

Comment:
Why do you turn the auto-refresh on here? If a user turns it off, I believe we shouldn't turn it on automatically after a bulk action.